pyiceberg/table/update/snapshot.py

```diff
@@ -445,6 +461,14 @@ def files_affected(self) -> bool:
         """Indicate if any manifest-entries can be dropped."""
         return len(self._deleted_entries()) > 0
 
+    def _validate(self, current_metadata: TableMetadata, Snapshot: Optional[Snapshot]) -> None:
+        if Snapshot is None:
+            raise ValueError("Snapshot cannot be None.")
```

Comment:
What do you think of:
```suggestion
            raise CommitFailedException("Snapshot cannot be None.")
```

Reply:
okay~ fixed